framework/src/userobject/ElementUserObject.C

```diff
@@ -52,3 +52,8 @@ ElementUserObject::ElementUserObject(const InputParameters & parameters) :
     addMooseVariableDependency(coupled_vars[i]);
 }
 
+bool
+ElementUserObject::requestedJacobian(unsigned int var) const
+{
+  return var < _requested_jacobian_flag.size() && _requested_jacobian_flag[var];
```

Comment:
Ummm... this is a bad idea.  Just use a `set`.  With thousands of variables this could be unnecessarily large... with just a few variables `set` is going to be fast anyway.